tests/test_state/test_storage_state.py

```diff
@@ -158,3 +301,294 @@ def test_clamp_energy_asserts_battery_traded_more_than_energy_per_slot(self, is_
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.pledged_buy_kWh, time_slot, 0)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.offered_sell_kWh, time_slot, 0)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.offered_buy_kWh, time_slot, 0)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.charge_history, time_slot, storage_state.initial_soc)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.charge_history_kWh, time_slot, storage_state.initial_capacity_kWh)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.energy_to_sell_dict, time_slot, 0)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.energy_to_buy_dict, time_slot, 0)
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.offered_history, time_slot, "-")
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.used_history, time_slot, "-")
+            assert_time_slot_in_dict_attribute_with_default_value(
+                storage_state.time_series_ess_share, time_slot, {ESSEnergyOrigin.UNKNOWN: 0.,
+                                                                 ESSEnergyOrigin.LOCAL: 0.,
+                                                                 ESSEnergyOrigin.EXTERNAL: 0.}
+            )
+
+    def test_delete_past_state_values_market_slot_not_in_past(self):
+        storage_state = StorageState()
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        with patch("gsy_e.gsy_e_core.util.ConstSettings.SettlementMarketSettings."
+                   "ENABLE_SETTLEMENT_MARKETS", True):
+            storage_state.delete_past_state_values(current_time_slot)
+            assert storage_state.pledged_sell_kWh.get(past_time_slot) is not None
+
+    def test_delete_past_state_values_market_slot_in_past(self):
+        storage_state = StorageState()
+        past_time_slot, current_time_slot, future_time_slots = self._initialize_time_slots()
+        active_market_slot_time_list = [past_time_slot, current_time_slot, *future_time_slots]
+        storage_state.add_default_values_to_state_profiles(active_market_slot_time_list)
+        with patch("gsy_e.gsy_e_core.util.ConstSettings.SettlementMarketSettings."
+                   "ENABLE_SETTLEMENT_MARKETS", False):
+            storage_state.delete_past_state_values(current_time_slot)
+            assert storage_state.pledged_sell_kWh.get(past_time_slot) is None
```

Comment:
Can you please explain? Shouldn't there entries for all provided `add_default_values_to_state_profiles` , independent of the ENABLE_SETTLEMENT_MARKETS setting?

Reply:
The methods here are supposed to test wether `delete_past_state_values()` method works properly. What it does is that it deletes key-value pairs from state profiles dicts (such as `self.pledge_sell_kWh`) if this are "in the past".

That is the reason for the names of these test methods `..._slot_not_in_past` and `..._slot_in_past`. 
The thing here is that "being in the past" (being a past state) can be relative depending on wether Settlement Markets are enabled or not.
When they are, then those market slots which happened less than an hour ago from `current_time_slot` are not past_states yet, and therefore should not be deleted when calling `delete_past_state_values()`. Therefore this lines:

```python
with patch("gsy_e.gsy_e_core.util.ConstSettings.SettlementMarketSettings."
             "ENABLE_SETTLEMENT_MARKETS", True):
      storage_state.delete_past_state_values(current_time_slot)
      assert storage_state.pledged_sell_kWh.get(past_time_slot) is not None
```

because `past_time_slot` is by construction only 15 minutes before current_time_slot, and therefore when the SettlementMarkets (SM) are enabled, is not technically a past state yet, and thus should not be deleted when calling the method.

On the other hand, if SM are disabled, then `past_time_slot` becomes indeed a past state and should therefore be deleted.

```python
with patch("gsy_e.gsy_e_core.util.ConstSettings.SettlementMarketSettings."
             "ENABLE_SETTLEMENT_MARKETS", False):
      storage_state.delete_past_state_values(current_time_slot)
      assert storage_state.pledged_sell_kWh.get(past_time_slot) is None
```

Reply:
Ok, thank you, so this is basically testing the utility function `is_time_slot_in_past_markets`, right?
Thanks a lot for the explanation, keep it as is then.